index.js

```diff
@@ -40,26 +41,30 @@ function Wayfarer (dft) {
   // match and call a route
   // (str, obj?) -> null
   function emit (route) {
-    assert.notEqual(route, undefined, "'route' must be defined")
+    var matched = match(route)
+
     var args = new Array(arguments.length)
+    args[0] = matched.params
     for (var i = 1; i < args.length; i++) {
       args[i] = arguments[i]
     }
 
-    var node = _trie.match(route)
-    if (node && node.cb) {
-      args[0] = node.params
-      var cb = node.cb
-      return cb.apply(cb, args)
-    }
+    return matched.cb.apply(matched.cb, args)
+  }
+
+  function match (route) {
+    assert.notEqual(route, undefined, "'route' must be defined")
+
+    var matched = _trie.match(route)
+    if (matched && matched.cb) return nodeToObject(matched)
 
     var dft = _trie.match(_default)
-    if (dft && dft.cb) {
-      args[0] = dft.params
-      var dftcb = dft.cb
-      return dftcb.apply(dftcb, args)
-    }
+    if (dft && dft.cb) return nodeToObject(dft)
 
     throw new Error("route '" + route + "' did not match")
   }
+
+  function nodeToObject (matched) {
+    return {cb: matched.cb, route: matched.cb.route, params: matched.params}
```

Comment:
heya, could we perhaps statically allocate the object ahead of time, and then reuse it on later runs? The router here might be called up to 60 times a second (sure hope that it's optimized tho, haha) - but that might be problematic later down the line.

so something along the lines of:
```js
var obj = {}

function nodeToObject (matched) {
  obj.cb = matched.cb,
  obj.route = matched.cb.route,
  obj.params = matched.params
  return obj
}
```

Reply:
The object creation isn't an issue at all. Chrome supports millions of operations per second. Stuff like iterating through keys and assigning them like it's done in the trie file is much slower. So I'd rather remove that one https://github.com/yoshuawuyts/wayfarer/blob/master/trie.js#L99

I don't like modifying an existing object. That will cause issues. Especially because I'd also like to pass that object around and still access the correct values in a next tick.

As alternative, thi should be much faster.
```
function Node (matched) {
  this.cb = matched.cb,
  this.route = matched.cb.route,
  this.params = matched.params
}
```

Reply:
Alright, yeah I'd be cool with that!